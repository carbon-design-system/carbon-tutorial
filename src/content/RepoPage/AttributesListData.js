import React, { Component } from 'react';

var ATTRIBUTE_SET_URL =
  'http://watsonpow01.rch.stglabs.ibm.com/services/attribute_detection_det/api/v1/attribute_sets/';

class AttributeList extends Component {
  constructor(props) {
    super(props);

    this.state = {
      loading: false,
      data: {},
    };
  }
  fetch() {
    var that = this;
    if (that.props.attributeSetId) {
      fetch(
        ATTRIBUTE_SET_URL +
          that.props.attributeSetId +
          '/attributes?_offset=1&_limit=500&show_disabled=true&include_base=false&version=2019-07-15&_fields=name&_fields=id&_fields=description&_fields=disabled&_fields=*'
      )
        .then(function(response) {
          return response.json();
        })
        .then(function(result) {
          that.setState({ data: result });
        });
    }
  }

  componentWillMount() {
    console.log('componentWillMount', this.state);
    this.setState({ loading: true });
    this.fetch();
    this.setState({ loading: false });
  }

  componentDidUpdate() {
    console.log('componentDidUpdate', this.state);
  }
  render = () => this.props.children(this.state);
}

export default AttributeList;
