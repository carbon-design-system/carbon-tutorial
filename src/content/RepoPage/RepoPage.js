import React from 'react';
const RepoPage = () => {
  return <div>REPO PAGE</div>;
};
export default RepoPage;
