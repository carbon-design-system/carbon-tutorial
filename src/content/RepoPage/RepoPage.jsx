import React from 'react';
const RepoPage = () => {
  return <div className="">Repo Page</div>;
};

export default RepoPage;
