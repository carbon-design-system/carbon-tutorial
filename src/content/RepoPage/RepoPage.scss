/*
 * RepoPage css File
*/

.RepoPage {
  /* propertys here */
}
