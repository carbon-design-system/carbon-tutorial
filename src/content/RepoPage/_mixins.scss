@use '@carbon/react/scss/spacing' as *;
@use '@carbon/react/scss/theme' as *;

@mixin repo-page-background() {
  background-color: $layer-01;
  position: relative;
}
