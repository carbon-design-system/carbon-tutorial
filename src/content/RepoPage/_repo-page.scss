.repo-page__r1 {
  padding-top: $spacing-05;
  padding-bottom: $spacing-05;
}
