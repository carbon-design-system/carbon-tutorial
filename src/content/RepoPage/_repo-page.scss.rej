diff a/src/content/RepoPage/_repo-page.scss b/src/content/RepoPage/_repo-page.scss	(rejected hunks)
@@ -2,3 +2,3 @@
-    padding-top: $spacing-05;
-    padding-bottom: $spacing-05;
-  }
\ No newline at end of file
+  padding-top: $spacing-05;
+  padding-bottom: $spacing-05;
+}
