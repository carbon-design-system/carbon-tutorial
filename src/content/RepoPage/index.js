import 'core-js/modules/es7.array.includes';
import 'core-js/modules/es6.array.fill';
import 'core-js/modules/es6.string.includes';
import 'core-js/modules/es6.string.trim';
import 'core-js/modules/es7.object.values';

import RepoPage from './RepoPage';
export default RepoPage;
