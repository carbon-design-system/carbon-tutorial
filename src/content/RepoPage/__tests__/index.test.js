// test file
import React from 'react';
import { shallow } from 'enzyme';
import RepoPage from '../RepoPage';

describe('Landing Page', () => {
  it('exists', () => {
    expect(shallow(<RepoPage />)).toMatchSnapshot();
  });
});
