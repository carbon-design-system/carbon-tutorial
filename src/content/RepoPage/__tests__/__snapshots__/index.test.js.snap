// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Page exists 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RepoPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <RepoQuery />,
      "className": "repo-page",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <RepoQuery />,
        "className": "repo-page",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
